### PR TITLE
Create Tag widget 1 template variation without title uppercase #1255

### PR DESCRIPTION
**Pull request by ithiame**

### CODE REVIEW

**Comment by coyotte508** [comment]
Let's use an additional "query" param instead

Eg `[TagProducts=slug?display=var-1 titleUppercase=false]`

**Comment by Tirodem** [comment]
> Let's use an additional "query" param instead
> 
> Eg `[TagProducts=slug?display=var-1 titleCase=regular]`
> 
> Then the main `TagWidget` can pass the appropriate class to the subwdigets (`"uppercase` or `""` for now, but later could be extended to `capitalize`, ...)

We have indeed a work to do there, refacto + cleaning.
For now, we have :
- ctaLess
- reverse
- noUC
- noBG
- maybe some more that I forgot

I don't know what the best, beginning the journey here or letting it go as it is and schedule a larger workshop on this in order to switch to tag widget templates variabilization.
BTW, if we change previous variations to parameters, we'll have to think about some kind of alias, for var-1-reverse-CtaLess to still work on existing CMS pages from current pilots and call var-1 with reverse & CtaLess parameters (this one combination precisely does not exist but you see the point I think).

**Comment by Tirodem** [comment]
That aside, it's working as it is ✅ 
@coyotte508 tell me what's the best, going like that or going with parameter before the wider task.

**Comment by coyotte508** [comment]
I think adding the param for this one makes sense - it easily applies to all widgets (while reverse is more complicated - maybe, because it could also be a single css class)

anyway let's start with this one

**Comment by ithiame** [comment]
`[Tag=shona-taine?display=var-1-reverse?titleCase=regular]`

File: src/lib/server/cms.ts [hunk 1]
```diff
@@ -86,7 +87,7 @@ export async function cmsFromContent(
 	const SLIDER_WIDGET_REGEX =
 		/\[Slider=(?<slug>[\p{L}\d_-]+)(?:[?\s]autoplay=(?<autoplay>[\d]+))?\]/giu;
 	const TAG_WIDGET_REGEX =
-		/\[Tag=(?<slug>[\p{L}\d_-]+)(?:[?\s]display=(?<display>[a-z0-9-]+))?\]/giu;
+		/\[Tag=(?<slug>[\p{L}\d_-]+)(?:[?\s]display=(?<display>[a-z0-9-]+))(?:[?\s]titleCase=(?<titleCase>[a-z0-9-]+))?\]/giu;
```
Comment (coyotte508) [inline src/lib/server/cms.ts:90]:
```suggestion
		/\[Tag=(?<slug>[\p{L}\d_-]+)(?:[?\s]display=(?<display>[a-z0-9-]+))(?:[?\s]titleCase=(?<titleCase>(upper|regular)))?\]/giu;
```

File: src/lib/components/TagWidget/TagWidgetVariation1Reverse.svelte [hunk 2]
```diff
@@ -22,7 +23,7 @@
 			class="mx-auto tagWidget tagWidget-main flex rounded sm:gap-2 {className} flex-row-reverse"
 		>
 			<div class="flex flex-col w-[50%] m-2">
-				<h2 class="text-6xl body-title pb-2 uppercase">{tag.title}</h2>
+				<h2 class="text-6xl body-title pb-2 {titleCase}">{tag.title}</h2>
```
Comment (coyotte508) [inline src/lib/components/TagWidget/TagWidgetVariation1Reverse.svelte:26]:
```suggestion
				<!-- possible casings: "uppercase", "normal-case" -->
				<h2 class="text-6xl body-title pb-2 {titleCase}">{tag.title}</h2>
```

Adding comment so tailwind can see what classes can be used

File: src/lib/server/cms.ts [hunk 3]
```diff
@@ -52,6 +52,7 @@ type TokenObject =
 			type: 'tagWidget';
 			slug: string;
 			display: string | undefined;
+			titleCase: string | undefined;
```
Comment (coyotte508) [inline src/lib/server/cms.ts:55]:
```suggestion
			titleCase: "regular" | "upper" | undefined;
```

File: src/lib/server/cms.ts
Comment (coyotte508) [inline src/lib/server/cms.ts:1]: Add:

```
export type CmsToken = Awaited<ReturnType<typeof cmsFromContent>>['tokens']['desktop'][number];
```

File: src/lib/components/TagWidget.svelte [hunk 4]
```diff
@@ -22,6 +22,7 @@
 	>;
 	export let pictures: Picture[];
 	export let displayOption = 'var-1';
+	export let titleCase = 'uppercase';
```
Comment (coyotte508) [inline src/lib/components/TagWidget.svelte:25]:
```suggestion
	export let titleCase: Extract<CmsToken, {type: "tagWidget"}>["titleCase"]; = 'upper';
	
	$: titleClassNames = titleCase === "upper" ? "uppercase" : "";
```

File: src/lib/components/TagWidget.svelte [hunk 5]
```diff
@@ -76,5 +78,6 @@
 		picture={pictures.find((picture) => picture.tag?.type === widget?.pictureType)}
 		avatar={pictures.find((picture) => picture.tag?.type === 'avatar')}
 		class={className}
+		{titleCase}
```
Comment (coyotte508) [inline src/lib/components/TagWidget.svelte:81]:
```suggestion
		{titleClassNames}
```

Apply to all widgets

File: src/lib/components/TagWidget/TagWidgetVariation1.svelte [hunk 6]
```diff
@@ -14,13 +14,14 @@
...
 	export let avatar: Picture | undefined;
+	export let titleCase = 'uppercase';
 </script>
 
 <div class="hidden sm:inline">
 	<div class="flex">
 		<div class="mx-auto tagWidget tagWidget-main flex rounded sm:gap-2 {className}">
 			<div class="flex flex-col w-[50%] m-2">
-				<h2 class="text-6xl body-title pb-2 uppercase">{tag.title}</h2>
+				<h2 class="text-6xl body-title pb-2 {titleCase}">{tag.title}</h2>
```
Comment (coyotte508) [inline src/lib/components/TagWidget/TagWidgetVariation1.svelte:24]:
```suggestion
				<h2 class="text-6xl body-title pb-2 {titleClassNames}">{tag.title}</h2>
```

apply to all widgets

File: src/lib/components/TagWidget/TagWidgetVariation1.svelte [hunk 7]
```diff
@@ -14,13 +14,14 @@
 	>;
 	export let picture: Picture | undefined;
 	export let avatar: Picture | undefined;
+	export let titleCase = 'uppercase';
```
Comment (coyotte508) [inline src/lib/components/TagWidget/TagWidgetVariation1.svelte:17]:
```suggestion
	export let titleClassNames = '';
```

**Review by coyotte508 — COMMENTED** [review]
> `[Tag=shona-taine?display=var-1-reverse?titleCase=regular]`

Note: we can use spaces past the first `?`:

`[Tag=shona-taine?display=var-1-reverse titleCase=regular]`

File: src/routes/(app)/tag/[id]/+page.svelte [hunk 8]
```diff
@@ -9,6 +9,13 @@
 		class="w-full rounded-xl body-secondPlan border-gray-300 border py-3 px-3 flex flex-col gap-2"
 	>
 		<TagWidget tag={data.tag} displayOption="var-1" pictures={data.pictures} />
+		<TagWidget
+			tag={data.tag}
+			displayOption="var-1"
+			titleClassName="regular"
```
Comment (coyotte508) [inline src/routes/(app)/tag/[id]/+page.svelte:15]: uhm no 

- CMS should pass `titleCase` to `TagWidget`
- `TagWidget` should pass `titleClassName` to its variations

check again my comments in the previous review

**Review by coyotte508 — APPROVED** [review]

**Pull request merged**